### ISSUE TITLE
Missing headers in homebrew install

### ISSUE THREAD

**Issue by coatless** [VowpalWabbit/vowpal_wabbit#1567]
When trying to compile using vowpalwabbit's header, provided by installing on macOS using homebrew, a fatal error occurs that branches out from the inclusion of `global_data.h`. The initial error was:

```bash
/usr/local/include/vowpalwabbit/global_data.h:20:10: 
fatal error: 'array_parameters.h' file not found
```

After copying the header file to location, there were then subsequent errors that arose with compilation due to missing header files.  In particular, the following header files required by including `global_data.h` were not found in `/usr/local/include/vowpalwabbit/`:

- `array_parameters.h`
- `array_parameters_dense.h`
- `hash.h` (under `/explore`)
- `no_label.h`
- `example_predict.h`
- `error_reporting.h`
- `parser_helper.h`

I think the reason for the header files missing is because these files are not retained in Makefile list:

https://github.com/JohnLangford/vowpal_wabbit/blob/64ac2f8a1f566bb35c811f4f4d2fc8243b5dc5f6/Makefile.am#L3-L8

Header files provided by homebrew: 

```bash
ls -la /usr/local/include/vowpalwabbit/
-rw-r--r--   1 ronin  admin   1997 Aug 12 20:17 action_score.h
-rw-r--r--   1 ronin  admin   7934 Aug 12 20:17 allreduce.h
-rw-r--r--   1 ronin  admin   1078 Aug 12 20:17 cb.h
-rw-r--r--   1 ronin  admin    320 Aug 12 20:17 cb_explore.h
-rw-r--r--   1 ronin  admin    833 Aug 12 20:17 comp_io.h
-rw-r--r--   1 ronin  admin   2525 Aug 12 20:17 config.h
-rw-r--r--   1 ronin  admin    513 Aug 12 20:17 constant.h
-rw-r--r--   1 ronin  admin   1228 Aug 12 20:17 cost_sensitive.h
-rw-r--r--   1 ronin  admin    472 Aug 12 20:17 crossplat_compat.h
-rw-r--r--   1 ronin  admin   4076 Aug 12 20:17 example.h
-rw-r--r--   1 ronin  admin  11166 Aug 12 20:17 ezexample.h
-rw-r--r--   1 ronin  admin  10430 Aug 12 20:17 feature_group.h
-rw-r--r--   1 ronin  admin    443 Aug 12 20:17 floatbits.h
-rw-r--r--   1 ronin  admin  18832 Aug 12 20:17 global_data.h
-rw-r--r--   1 ronin  admin   9137 Aug 12 20:17 io_buf.h
-rw-r--r--   1 ronin  admin    821 Aug 12 20:17 label_parser.h
-rw-r--r--   1 ronin  admin  18462 Aug 12 20:17 learner.h
-rw-r--r--   1 ronin  admin   1521 Aug 12 20:17 loss_functions.h
-rw-r--r--   1 ronin  admin   2226 Aug 12 20:17 memory.h
-rw-r--r--   1 ronin  admin    653 Aug 12 20:17 multiclass.h
-rw-r--r--   1 ronin  admin    496 Aug 12 20:17 multilabel.h
-rw-r--r--   1 ronin  admin    705 Aug 12 20:17 parse_example.h
-rw-r--r--   1 ronin  admin   3689 Aug 12 20:17 parse_primitives.h
-rw-r--r--   1 ronin  admin   2379 Aug 12 20:17 parser.h
-rw-r--r--   1 ronin  admin    573 Aug 12 20:17 simple_label.h
-rw-r--r--   1 ronin  admin   6367 Aug 12 20:17 v_array.h
-rw-r--r--   1 ronin  admin   7424 Aug 12 20:17 v_hashmap.h
-rw-r--r--   1 ronin  admin   6400 Aug 12 20:17 vw.h
-rw-r--r--   1 ronin  admin   2766 Aug 12 20:17 vw_exception.h
-rw-r--r--   1 ronin  admin    423 Aug 12 20:17 vw_validate.h
-rw-r--r--   1 ronin  admin   5814 Aug 12 20:17 vwdll.h
```

**Comment by lokitoth** [comment]
@jackgerrits You mentioned that Homebrew is working again? Is this issue still current?

**Comment by jackgerrits** [comment]
Hi @coatless, sorry it has taken so long to get to this. You're right, the required headers are not part of the group that is installed and this issue persisted through the transition to CMake. The solution is to add those headers to the install, and I'll submit a PR to resolve this.

I would like to add though that traditionally VW has been a command line tool first, so the interface has primarily been maintained about that. There hasn't really been strict guarantees around the library interface which resulted in the issue you saw. So using the headers is a bit of use at your own risk at the moment as I'm sure you've noticed over time.

We plan on changing this for VW 9 though. We plan on defining a proper public interface so users can feel more sure about compatability on the library interface.

**Comment by coatless** [comment]
@jackgerrits thanks! This would be great and help us with moving `rvw`, the _R_ interface into Vowpal Wabbit, onto CRAN.

**Comment by jackgerrits** [comment]
I'm going to close this as it should be fixed in the repo, we plan to do another minor release in the soonish future. When that's out the homebrew package will be updated and you should be unblocked.
